driver/youki.go

```diff
@@ -0,0 +1,255 @@
+package driver
+
+import (
+	"bufio"
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"strings"
+	"time"
+
+	"github.com/estesp/bucketbench/utils"
+	log "github.com/sirupsen/logrus"
+)
+
+const defaultYoukiBinary = "youki"
+
+// YoukiDriver is an implementation of the driver interface for Youki.
+// IMPORTANT: This implementation does not protect instance metadata for thread safely.
+// At this time there is no understood use case for multi-threaded use of this implementation.
+type YoukiDriver struct {
+	youkiBinary string
+}
+
+// YoukiContainer is an implementation of the container metadata needed for youki
+type YoukiContainer struct {
+	name       string
+	bundlePath string
+	detached   bool
+	state      string
+	pid        string
+	trace      bool
+}
+
+// NewYoukiDriver creates an instance of the youki driver, providing a path to youki
+func NewYoukiDriver(binaryPath string) (Driver, error) {
+	if binaryPath == "" {
+		binaryPath = defaultYoukiBinary
+	}
+	resolvedBinPath, err := utils.ResolveBinary(binaryPath)
+	if err != nil {
+		return &YoukiDriver{}, err
+	}
+	driver := &YoukiDriver{
+		youkiBinary: resolvedBinPath,
+	}
+	return driver, nil
+}
+
+// newYoukiContainer creates the metadata object of a youki-specific container with
+// bundle, name, and any required additional information
+func newYoukiContainer(name, bundlepath string, detached bool, trace bool) Container {
+	return &YoukiContainer{
+		name:       name,
+		bundlePath: bundlepath,
+		detached:   detached,
+		trace:      trace,
+	}
+}
+
+// Name returns the name of the container
+func (c *YoukiContainer) Name() string {
+	return c.name
+}
+
+// Detached returns whether the container should be started in detached mode
+func (c *YoukiContainer) Detached() bool {
+	return c.detached
+}
+
+// Trace returns whether the container should be started with tracing enabled
+func (c *YoukiContainer) Trace() bool {
+	return c.trace
+}
+
+// Image returns the bundle path that youki will use
+func (c *YoukiContainer) Image() string {
+	return c.bundlePath
+}
+
+// Command is not implemented for the youki driver type
+// as the command is embedded in the config.json of the rootfs
+func (c *YoukiContainer) Command() string {
+	return ""
+}
+
+// Pid returns the process ID in cases where this container instance is
+// wrapping a potentially running container
+func (c *YoukiContainer) Pid() string {
+	return c.pid
+}
+
+// State returns the queried state of the container (if available)
+func (c *YoukiContainer) State() string {
+	return c.state
+}
+
+//GetPodID return pod-id associated with container.
+//only used by CRI-based drivers
+func (c *YoukiContainer) GetPodID() string {
```

Comment:
As mentioned in the other driver, same issue here in lines 98-99 with the space required after the `//`; thanks